### DISCUSSION TITLE
Extracting IfcStair vertices

### DISCUSSION THREAD

**Discussion by DprOliv** [IfcOpenShell/IfcOpenShell#5605] [General]
Hello 

Trying here ot extract the vertices of a stair...
I tried the very same approach used to explore spaes vertices and even staitrs
Which is:
geom_sets = ifcGeom.settings()
list_stairs = ifcModel.by_type('ifcStair')
for obj in list_stairs:
         obj_Shp = ifcGeom.create_shape(geom_sets, obj)
         verts = obj_Shp.geometry.verts

And it's presenting an erros... and the dometri

As i mentioned...
The very same code was used to extract vertices of spaces geometries...

is that because stairs are in fact compositions of other entities such as slabs... ?

Regards to all

**Comment by aothms** [comment]
> is that because stairs are in fact compositions of other entities such as slabs... ?

Most likely.

```
for obj in list_stairs:
  for rel in obj.IsDecomposedBy:
    for part in rel.RelatedObjects:
      ifcGeom.create_shape(geom_sets, part)
```

**Reply by DprOliv** [reply]
Perfect. Usage of this Decompose relation as you suggested, solved the problem.
Thank you @aothms !
